### DISCUSSION TITLE
C or C++ API for Mujoco Tasks?

### DISCUSSION THREAD

**Discussion by skellco** [Farama-Foundation/Gymnasium#1181] [General]
We develop RL algorithms in C++ and would like to evaluate them on the Gymnasium Mujoco environments. Are there any example C or C++ APIs for these tasks?

**Comment by Kallinteris-Andreas** [comment]
Unfortunately, there is no direct way to interface with gymnasium (or any other in the Farama Project) using a C++ API directly

fortunately the core Gymnasium API uses NumPy for the function's arguments and return values (`step()`, `reset()`), so there is likely some way to interface with C++,

Though I can not really help you with that, a web search is your friend

And if you find a solution please, share it here (you would be the first to use gymnasium's environments in C++ to my knowledge)

I would probably start here: https://www.boost.org/doc/libs/1_86_0/libs/python/doc/html/tutorial/tutorial/embedding.html
